src/full_node.py

```diff
@@ -489,7 +499,9 @@ def _shutdown(self):
                 start = time.time()
                 async with self.store.lock:
                     # The block gets permanantly added to the blockchain
-                    result = await self.blockchain.receive_block(block)
+                    validated, pos = prevalidate_results[index]
+                    index += 1
+                    result = await self.blockchain.receive_block(block, validated, pos)
```

Comment:
Also, we might benefit from calling prevalidate_blocks before the other call to receive_block, even though we're only doing one at a time, right? (because it will validate it in a new process, and full node can still do other things).

Reply:
sure